lib/Controller/UsersController.php

```diff
@@ -199,4 +171,50 @@ public function get(string $userId): DataResponse {
 
 		return new DataResponse($guests);
 	}
+
+	private function getStatusMessage(string $status): string {
+		return match ($status) {
+			Transfer::STATUS_WAITING => $this->l10n->t('Waiting'),
+			Transfer::STATUS_STARTED => $this->l10n->t('Started'),
+		};
+	}
+
+	/**
+	 * Transfer guest to a full account
+	 *
+	 * @param string $email Guest email
+	 * @param string $userId User id for full account
+	 */
+	public function transfer(string $email, string $userId): DataResponse {
+		$author = $this->userSession->getUser();
+		if (!($author instanceof IUser)) {
+			return new DataResponse([], Http::STATUS_UNAUTHORIZED);
+		}
+
+		$sourceUser = $this->userManager->get($email);
+		if (!($sourceUser instanceof IUser)) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		if ($this->userManager->userExists($userId)) {
+			return new DataResponse([
+				'message' => $this->l10n->t('User already exists')
+			], Http::STATUS_CONFLICT);
+		}
+
+		if (!$this->guestManager->isGuest($sourceUser)) {
+			return new DataResponse([], Http::STATUS_CONFLICT);
+		}
+
+		$transfer = $this->transferService->getTransfer($sourceUser, $userId);
+		if ($transfer instanceof Transfer) {
+			return new DataResponse([
```

Comment:
What kind of response do we have if a transfer is requested with a different `$sourceUser2` but `$userId` with an existing pending transfer?